DatadogSessionReplay/Sources/DatadogSessionReplay/DatadogCoreIntegration/RUMContextReceiver.swift

```diff
@@ -85,12 +87,18 @@ private extension FeatureBaggage {
     var rumContext: RUMContext? {
         guard let applicationID: String = self[RUMDependency.applicationIDKey],
               let sessionID: String = self[RUMDependency.sessionIDKey],
-              let viewID: String = self[RUMDependency.viewIDKey]
+              let viewID: String = self[RUMDependency.viewIDKey],
+              let serverTimeOffset: TimeInterval = self[RUMDependency.serverTimeOffset]
         else {
             // Current RUM session is not sampled
             return nil
         }
 
-        return RUMContext(applicationID: applicationID, sessionID: sessionID, viewID: viewID)
+        return RUMContext(
+            applicationID: applicationID,
+            sessionID: sessionID,
+            viewID: viewID,
+            serverTimeOffset: Int64(serverTimeOffset * 1_000)
```

Comment:
This may crash with overflow. We shouldn't make any assumption on `serverTimeOffset` being always "realistic" - I don't recall any upstream logic that will guard that.

Let's use [`Int64(withNoOverflow:)`](https://github.com/DataDog/dd-sdk-ios/blob/4f36c8a23f4416237d8b71846caab747c75b1e68/session-replay/Sources/DatadogSessionReplay/Utilities/SwiftExtensions.swift#L13) instead.